replica/table.cc

```diff
@@ -3011,6 +3029,18 @@ table::mutation_query(schema_ptr s,
         co_return reconcilable_result();
     }
 
+    // 1 for the table and >= 1 for the compaction groups that overlap
+    utils::small_vector<gate::holder, 2> async_gate_holders;
+    async_gate_holders.push_back(_async_gate.hold());
+    {
```

Comment:
why the nested block?

Reply:
To free `token_ranges`, it is not required after checking the overlaps.